tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -755,7 +755,7 @@ static bool passCursorInfoForDecl(const ValueDecl *VD,
         PO.SkipIntroducerKeywords = true;
         PO.ArgAndParamPrinting = PrintOptions::ArgAndParamPrintingMode::ArgumentOnly;
         XMLEscapingPrinter Printer(RelDeclsStream);
-        if (BaseType) {
+        if (BaseType && RelatedDecl->getDeclContext() == BaseType->getAnyNominal()) {
```

Comment:
We cannot get rid of the base type altogether because we may print other overloads of the original decl. These overloads need substitutions as well. I think checking whether the related decl is in a type context achieves the right balance between correctness and complexity.